plugin/core/transports.py

```diff
@@ -256,14 +289,24 @@ def start_subprocess() -> subprocess.Popen:
             sock = _connect_tcp(config.tcp_port)
             if sock is None:
                 raise RuntimeError("Failed to connect on port {}".format(config.tcp_port))
-            reader = sock.makefile('rwb')  # type: ignore
-            writer = reader
+            reader = writer = sock.makefile('rwb')
+        elif config.node_ipc:
+            reader = writer = config.node_ipc.parent_connection  # type: ignore
         else:
-            reader = process.stdout  # type: ignore
-            writer = process.stdin  # type: ignore
+            if not process.stdout or not process.stdin:
```

Comment:
this raise Exception can be deleted
there is check down bellow https://github.com/sublimelsp/LSP/pull/2015/files#diff-2abf74ab722813d8b12a5da0a054ac7ba7ee95f47278607834d5dc672357b818R306

Reply:
check mypy